docs/src/docs/asciidoc/user/migration-guide.adoc

```diff
@@ -0,0 +1,37 @@
+---
+---
+= Migration Guide
+ifndef::sourcedir32[]
+include::common.adoc[]
+endif::sourcedir32[]
+
+ifdef::notBuildingForSite[]
+include::menu.adoc[]
+endif::notBuildingForSite[]
+
+== Introduction
+
+This guide provide sample code snippets to help migrate ehcache2 code to ehcache3 code.
+
+
+== Custom Expiry
+
+Ehcache2 Code snippet:
```

Comment:
That's really terse. I would add introductory paragraph explaining the goal of this section and the describe a little bit the example in text before diving into code.